ansible_anonymizer/anonymizer.py

```diff
@@ -298,12 +298,16 @@ def hide_user_name(block: str) -> str:
     }
 
     def _rewrite(m: re.Match[str]) -> str:
-        user = m.group("user_name") if m.group("user_name") in known_users else "ano-user"
+        user = (
+            m.group("user_name")
+            if (m.group("user_name") in known_users or is_jinja2_expression(m.group("user_name")))
+            else "ano-user"
+        )
         return m.group("before") + user
 
     user_regexes = [
-        r"(?P<before>[c-z]:\\users\\)(?P<user_name>\w{,255})",
-        r"(?P<before>/(home|Users)/)(?P<user_name>[a-z0-9_-]{,255})",
+        r"(?P<before>[c-z]:\\users\\)(?P<user_name>([a-z0-9_-]|{{\s*.*?\s*}})\w{,255})",
+        r"(?P<before>/(home|Users)/)(?P<user_name>([a-z0-9_-]|{{\s*.*?\s*}})[a-z0-9_-]{,255})",
```

Comment:
Why is `[a-z0-9_-]` for? We've got already a pattern to match the login name.

Reply:
If I dont put it with an or (|), it will match the jinjas but not the regular texts: 

https://regex101.com/r/y2o0d9/1

However, upon your comment the second one seems redundant. removing it seems to be working: https://regex101.com/r/54R0ZI/1

Let me check with the code and update it if the tests work.

Reply:
On Windows, a login name can come with non ASCII characters. `[a-z0-9_-]` is more strict than just `\w` which accepts any Unicode characters.

```
>>> import re
>>> re.match(r"[a-z0-9_-]+", "aaa")
<re.Match object; span=(0, 3), match='aaa'>
>>> re.match(r"[a-z0-9_-]+", "ááá")
```

Reply:
> On Windows, a login name can come with non ASCII characters. `[a-z0-9_-]` is more strict than just `\w` which accepts any Unicode characters.
> 
> ```
> >>> import re
> >>> re.match(r"[a-z0-9_-]+", "aaa")
> <re.Match object; span=(0, 3), match='aaa'>
> >>> re.match(r"[a-z0-9_-]+", "ááá")
> ```

@goneri I must have preserved that `\w` for the windows version. Note that the code updates are different than the above regex links, which I created before I understood your point better.